implementations/app-kotlin/src/__tests__/cases/000-sanity/output/types.kt

```diff
@@ -0,0 +1,241 @@
+    val foo: String,
+)
+
+@Serializable
+data class Else(
+    val _else: String,
+)
+
+/// Objects END ///
+
+/// Enums START ///
+@Serializable
+enum class CustomEnum {
+    STRING,
+    BYTES
+}
+
+@Serializable
+enum class While {
+    _for,
+    _in
+}
+
+/// Enums END ///
+
+/// Imported Objects START ///
+/* URI: "testimport.uri.eth" */
+@Serializable
+data class TestImportObject(
+    val _object: TestImportAnotherObject,
+    val optObject: TestImportAnotherObject? = null,
+    val objectArray: List<TestImportAnotherObject>,
+    val optObjectArray: List<TestImportAnotherObject?>? = null,
+    val en: TestImportEnum,
+    val optEnum: TestImportEnum? = null,
+    val enumArray: List<TestImportEnum>,
+    val optEnumArray: List<TestImportEnum?>? = null,
+)
+
+/* URI: "testimport.uri.eth" */
+@Serializable
+data class TestImportAnotherObject(
+    val prop: String,
+)
+
+/// Imported Objects END ///
+
+/// Imported Enums START ///
+/* URI: "testimport.uri.eth" */
+@Serializable
+enum class TestImportEnum {
+    STRING,
+    BYTES
+}
+
+/* URI: "testimport.uri.eth" */
+@Serializable
+enum class TestImportEnumReturn {
+    STRING,
+    BYTES
+}
+
+/// Imported Enums END ///
+
+/// Imported Envs START ///
+@Serializable
+data class TestImportEnv(
+    val _object: TestImportAnotherObject,
+    val optObject: TestImportAnotherObject? = null,
+    val objectArray: List<TestImportAnotherObject>,
+    val optObjectArray: List<TestImportAnotherObject?>? = null,
+    val en: TestImportEnum,
+    val optEnum: TestImportEnum? = null,
+    val enumArray: List<TestImportEnum>,
+    val optEnumArray: List<TestImportEnum?>? = null,
+)
+
+/// Imported Envs END ///
+
+/// Imported Modules START ///
+/* URI: "testimport.uri.eth" */
+@Serializable
+data class TestImportArgsImportedMethod(
+    val str: String,
+    val optStr: String? = null,
+    val u: UInt,
+    val optU: UInt? = null,
+    val uArrayArray: List<List<UInt?>?>,
+    val _object: TestImportObject,
+    val optObject: TestImportObject? = null,
+    val objectArray: List<TestImportObject>,
+    val optObjectArray: List<TestImportObject?>? = null,
+    val en: TestImportEnum,
+    val optEnum: TestImportEnum? = null,
+    val enumArray: List<TestImportEnum>,
+    val optEnumArray: List<TestImportEnum?>? = null,
+)
+
+/* URI: "testimport.uri.eth" */
+@Serializable
+data class TestImportArgsAnotherMethod(
+    val arg: List<String>,
+)
+
+/* URI: "testimport.uri.eth" */
+@Serializable
+data class TestImportArgsReturnsArrayOfEnums(
+    val arg: String,
+)
+
+/* URI: "testimport.uri.eth" */
+open class TestImport(
+    client: Invoker? = null,
+    env: TestImportEnv? = null,
+    uri: Uri? = null
+) {
+    companion object {
+        val uri: Uri = Uri("testimport.uri.eth")
+    }
+
+    protected val defaultClient: Invoker by lazy {
+        polywrapClient { addDefaults() }
+    }
+    protected val defaultUri: Uri? = null
+    protected val defaultEnv: TestImportEnv? = null
```

Comment:
As a non Kotlin dev, I am just curious why we aren't using same pattern as other clients where we use constructor and abstract methods that can be extended

Reply:
In Kotlin, abstract members can only be in an abstract class. If a user wants to, they can extend the class and override the members `defaultClient`, `defaultUri`, and `defaultEnv`